packages/shared/lib/services/connection.service.ts

```diff
@@ -1042,10 +1042,18 @@ class ConnectionService {
     // Parses and arbitrary object (e.g. a server response or a user provided auth object) into AuthCredentials.
     // Throws if values are missing/missing the input is malformed.
     public parseRawCredentials(rawCredentials: object, authMode: AuthModeType, template?: ProviderOAuth2 | ProviderTwoStep): AuthCredentials {
-        const rawCreds = rawCredentials as Record<string, any>;
+        let rawCreds = rawCredentials as Record<string, any>;
 
         switch (authMode) {
             case 'OAUTH2': {
+                const isSlackUserToken = !rawCreds['access_token'] && !!rawCreds['authed_user'];
```

Comment:
I think it would make more sense to rely on the template name than on the shape of the token response.